src/gt4py/cartesian/frontend/gtscript_frontend.py

```diff
@@ -1417,9 +1417,14 @@ def visit_Assign(self, node: ast.Assign) -> list:
         for t in node.targets[0].elts if isinstance(node.targets[0], ast.Tuple) else node.targets:
             name, spatial_offset, data_index = self._parse_assign_target(t)
             if spatial_offset:
```

Comment:
are these the only changes to make it work? that's nice. It means we "only" have to sort out the conceptual problems

Reply:
I was expecting more too! So far though, it's all it takes. Next I'll put in some integration tests with conditionals and I'll simplify Oliver's dual for-loop example he pulled from SHiELD microphysics to bring something we've encountered in the wild.